Source/HYPFormsManager.m

```diff
@@ -556,69 +563,114 @@ - (NSArray *)updateTargets:(NSArray *)targets
-                        [self.values setObject:[value stringValue] forKey:field.fieldID];
-                        [values addEntriesFromDictionary:@{fieldID : [value stringValue]}];
+                } else if (value) {
+                    if ([value isKindOfClass:[NSString class]]) {
+                        if ([value length] == 0) value = defaultEmptyValue;
+                        [values addEntriesFromDictionary:@{fieldID : value}];
                     } else {
-                        [self.values setObject:@"" forKey:field.fieldID];
-                        [values addEntriesFromDictionary:@{fieldID : defaultEmptyValue}];
+                        if ([value respondsToSelector:NSSelectorFromString(@"stringValue")]) {
+                            [self.values setObject:[value stringValue] forKey:field.fieldID];
+                            [values addEntriesFromDictionary:@{fieldID : [value stringValue]}];
+                        } else {
+                            [self.values setObject:@"" forKey:field.fieldID];
+                            [values addEntriesFromDictionary:@{fieldID : defaultEmptyValue}];
+                        }
                     }
+                } else {
+                    [values addEntriesFromDictionary:@{fieldID : defaultEmptyValue}];
                 }
-            } else {
-                [values addEntriesFromDictionary:@{fieldID : defaultEmptyValue}];
             }
-        }
 
-        id result = [field.formula hyp_runFormulaWithValuesDictionary:values];
-        field.fieldValue = result;
+            id result = [field.formula hyp_runFormulaWithValuesDictionary:values];
+            field.fieldValue = result;
 
-        if (result) {
-            [self.values setObject:result forKey:field.fieldID];
-        } else {
-            [self.values removeObjectForKey:field.fieldID];
+            if (result) {
+                [self.values setObject:result forKey:field.fieldID];
+            } else {
+                [self.values removeObjectForKey:field.fieldID];
+            }
         }
     }
 
     return updatedIndexPaths;
 }
 
+- (NSArray *)enableTargets:(NSArray *)targets
+{
+    return [self updateTargets:targets withEnabled:YES];
+}
+
+- (NSArray *)disableTargets:(NSArray *)targets
+{
+    return [self updateTargets:targets withEnabled:NO];
+}
+
+- (NSArray *)updateTargets:(NSArray *)targets withEnabled:(BOOL)enabled
+{
+    NSMutableArray *indexPaths = [NSMutableArray new];
+
+    for (HYPFormTarget *target in targets) {
+        if (target.type == HYPFormTargetTypeSection) continue;
+        if ([self.hiddenFieldsAndFieldIDsDictionary objectForKey:target.targetID]) continue;
+
+        [self fieldWithID:target.targetID includingHiddenFields:YES completion:^(HYPFormField *field, NSIndexPath *indexPath) {
+            if (field) {
+                field.disabled = !enabled;
+                if (indexPath) [indexPaths addObject:indexPath];
```

Comment:
Since you create the indexPaths array on line 657, check if it isn't nil feels like an overkill here.